File: ios/TriPay/Domain/Models/SyncMetadata.swift
import Foundation

enum SyncStatus: String, Codable {
    case localOnly = "LocalOnly"
    case synced = "Synced"
    case conflict = "Conflict"
    case pendingSync = "PendingSync"
}

struct SyncMetadata: Codable {
    var lastSyncedAt: Date?
    var syncStatus: SyncStatus
    var version: Int
    var deviceId: String
}

File: android/app/src/main/kotlin/com/tripay/domain/models/SyncMetadata.kt
package com.tripay.domain.models

import java.util.Date

enum class SyncStatus {
    LOCAL_ONLY,
    SYNCED,
    CONFLICT,
    PENDING_SYNC
}

data class SyncMetadata(
    val lastSyncedAt: Date?,
    val syncStatus: SyncStatus,
    val version: Int,
    val deviceId: String
)
